File: todo_frontend/lib/todo_database.dart
import 'dart:convert';
import 'package:http/http.dart' as http;

class TodoDatabase {
  static Future<List<dynamic>> fetchTasks() async {
    final response = await http.get(Uri.parse('http://localhost:3000/tasks'));
    if (response.statusCode == 200) {
      return json.decode(response.body);
    } else {
      throw Exception('Failed to load tasks');
    }
  }

  static Future<void> addTask(String title, String description) async {
    await http.post(
      Uri.parse('http://localhost:3000/tasks'),
      headers: {"Content-Type": "application/json"},
      body: json.encode({'title': title, 'description': description}),
    );
  }

  static Future<void> updateTask(int id, bool isCompleted) async {
    await http.put(
      Uri.parse('http://localhost:3000/tasks/$id'),
      headers: {"Content-Type": "application/json"},
      body: json.encode({'is_completed': isCompleted ? 1 : 0}),
    );
  }

  static Future<void> deleteTask(int id) async {
    await http.delete(Uri.parse('http://localhost:3000/tasks/$id'));
  }
}

File: todoapp_frontend/src/todo_database.js
// Fungsi untuk mengambil semua tugas
export const fetchTasks = async () => {
    try {
      const response = await fetch('http://localhost:4000/tasks'); // Pastikan URL sesuai dengan backend Anda
      if (response.ok) {
        const data = await response.json();
        return data;
      } else {
        throw new Error('Gagal memuat tugas');
      }
    } catch (error) {
      console.error('Error:', error);
    }
  };
  
  // Fungsi untuk menambahkan tugas baru
  export const addTask = async (title, description) => {
    try {
      await fetch('http://localhost:4000/tasks', {
        method: 'POST',
        headers: { 'Content-Type': 'application/json' },
        body: JSON.stringify({ title, description }),
      });
    } catch (error) {
      console.error('Error:', error);
    }
  };
  
  // Fungsi untuk memperbarui status tugas
  export const updateTask = async (id, isCompleted) => {
    try {
      await fetch(`http://localhost:4000/tasks/${id}`, {
        method: 'PUT',
        headers: { 'Content-Type': 'application/json' },
        body: JSON.stringify({ is_completed: isCompleted ? 1 : 0 }),
      });
    } catch (error) {
      console.error('Error:', error);
    }
  };
  
  // Fungsi untuk menghapus tugas
  export const deleteTask = async (id) => {
    try {
      await fetch(`http://localhost:4000/tasks/${id}`, {
        method: 'DELETE',
      });
    } catch (error) {
      console.error('Error:', error);
    }
  };
  